### PR TITLE
fix: Partition type hint and scheme-prefixed output_path mangling

### PR DESCRIPTION
**Pull request by ceholden**
## Technical Context

- **Related Issues:** N/A but happy to open if that helps!
- **Breaking Changes:** No

## Description

This PR addresses two issues I had when using `stac-geoparquet` to dump our PgSTAC catalog into STAC GeoParquet.

1. The type hint on `Partition.[start | end]` should be nullable since Collections in PgSTAC without a `partition_trunc` will have an `[-infinity, infinity]` date range, causing the values to both be `None`.
    - Fix: c844bf89e666e9d1f3452223275c5d5a00292369
2. A scheme-prefixed `output_path` given to `to_parquet` is mangled by `Path()` when a `filesystem` is passed.
    - This wasn't captured in existing tests (no scheme-prefixed test data like `file://` or `s3://`), so I added a new regression test. Happy to merge if that is preferred.
    - This may not have been caught before because it requires setting _both_ `output_path` as scheme-prefixed **and** passing a `filesystem`. Alternatively someone could have provided a `filesystem` and an `output_path` like `"bucket/key.parquet"`
    - Regression test: 79cbac57b91b9697631411d645b99ee91dbc4731
    - Fix: 5b751789f257eabf130cee876e9f14731fd2ca5e

---

## Checklist

- [x] **Linting:** Code is formatted and linted
- [x] **Tests:** Tests pass. I have included new tests for these changes where applicable.
- [x] **Edge Cases:** I have manually verified "unhappy paths" and edge cases beyond the basic success criteria (e.g., database connection timeouts, malformed input, strict mapping rejections).
- [x] **Documentation:** I have updated `README.md` to reflect any new environment variables, configuration changes, or breaking updates.
- [x] **Accountability:** I can explain the implementation logic for every line of code submitted.

---

## AI tool usage

- [x] AI (Copilot or something similar) supported my development of this PR. See our [policy about AI tool use](https://stac-utils.github.io/ai-contribution-policy). Use of AI tools *must* be indicated.
    - I had Claude prototype regression test that I rewrote

---

> **Policy:** We require a "human-in-the-loop." You are the author and are fully accountable for all submitted code. Please ensure all tool-generated content is thoroughly reviewed before submission to ensure it is not an "extractive contribution" that squanders maintainer time.

### CODE REVIEW

File: stac_geoparquet/arrow/_util.py [hunk 1]
```diff
@@ -12,6 +13,19 @@
...
 
+def resolve_output_path_and_filesystem(
+    output_path: str | Path,
+    filesystem: pa.fs.FileSystem | None,
+) -> tuple[pa.fs.FileSystem, str | Path]:
+    """Resolve an (output_path, filesystem) pair, normalizing scheme-prefixed URIs."""
+    if filesystem is None:
+        return pa.fs.FileSystem.from_uri(output_path)
+    if isinstance(output_path, str) and "://" in output_path:
+        _, filepath = pa.fs.FileSystem.from_uri(output_path)
```
Comment (gadomski) [inline stac_geoparquet/arrow/_util.py]: If the `output_path` is resolved to a different filesystem than `filesystem`, this _could_ be surprising to the user? Especially since we return the original filesystem.

Comment (ceholden) [reply stac_geoparquet/arrow/_util.py]: The `filepath` returned from `from_uri` would not include the filesystem scheme (it's basically a glorified `urlsplit`, more on that below), so it is agnostic of the filesystem.

```python
>>> cases = ["gs://bucket/foo", "file:///test", "s3://bucket/buz"]
>>> [FileSystem.from_uri(case)[1] for case in cases]
['bucket/foo', '/test', 'bucket/buz']  # no filesystem detail present
```

Having poked this a bit more, I realized `FileSystem.from_uri` for this is very heavy handed and can involve side effects. For example, `FileSystem.from_uri("s3://bucket/key")` will do a network request, and if the bucket doesn't exist it can error. I'm pushing a commit to switch to `urllib.parse.urlsplit` to strip the scheme, which does what `from_uri` does but much simpler.

File: stac_geoparquet/arrow/_util.py [hunk 2]
```diff
@@ -2,16 +2,28 @@
...
+    output_path: str | Path,
+    filesystem: pa.fs.FileSystem | None,
+) -> tuple[pa.fs.FileSystem, str]:
+    """Resolve an (output_path, filesystem) pair, normalizing scheme-prefixed URIs."""
+    if filesystem is None:
+        return pa.fs.FileSystem.from_uri(output_path)
+    if isinstance(output_path, str) and "://" in output_path:
+        parsed = urlsplit(output_path)
+        return filesystem, f"{parsed.netloc}{parsed.path}"
+    return filesystem, str(output_path)
```
Comment (gadomski) [inline stac_geoparquet/arrow/_util.py:24]: `filesystem` has now become a pass-through argument, unused other than the `is None` check, which implies that check should be pushed to the call site and removed from this function?

**Review by gadomski — CHANGES_REQUESTED** [review]

File: stac_geoparquet/arrow/_util.py (hunk 2 above)
Comment (ceholden) [reply stac_geoparquet/arrow/_util.py:24]: Sure, can do that. It will just duplicate the `filesystem is None` check in 3 places

Comment (gadomski) [reply stac_geoparquet/arrow/_util.py:24]: Right — if you want to keep that logic, I'd recommend changing the argument to `resolve_from_uri` or something — there's no reason to add `FileSystem` to the function args if we only need a boolean.

Comment (gadomski) [reply stac_geoparquet/arrow/_util.py:24]: Ah, never mind, I'm re-reading it and seems fine. Disgrard comment.

**Review by gadomski — APPROVED** [review]

File: stac_geoparquet/arrow/_util.py (hunk 2 above)
Comment (ceholden) [reply stac_geoparquet/arrow/_util.py:24]: Thanks for speedy reviews/re-reviews! 🙇

Side thought - I'm hoping the filesytem/filepath resolution won't ever need to be more complex, but splitting that out in this function might be useful if that needs to happen
